jetbrains-core/tst/software/aws/toolkits/jetbrains/ui/feedback/FeedbackTest.kt

```diff
@@ -0,0 +1,79 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.ui.feedback
+
+import com.intellij.openapi.ui.ValidationInfo
+import com.intellij.testFramework.ProjectRule
+import com.intellij.testFramework.runInEdtAndWait
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Rule
+import org.junit.Test
+import software.amazon.awssdk.services.toolkittelemetry.model.Sentiment
+import software.aws.toolkits.resources.message
+
+class FeedbackTest {
+    @Rule
+    @JvmField
+    val projectRule = ProjectRule()
+
+    @Test
+    fun panelInitiallyNegative() {
+        val panel = SubmitFeedbackPanel(initiallyPositive = false)
+        assertThat(panel.sentiment).isEqualTo(Sentiment.NEGATIVE)
+    }
+
+    @Test
+    fun panelInitiallyPositive() {
+        val panel = SubmitFeedbackPanel(initiallyPositive = true)
+        assertThat(panel.sentiment).isEqualTo(Sentiment.POSITIVE)
+    }
+
+    @Test
+    fun noSentimentSet() {
+        runInEdtAndWait {
+            val dialog = FeedbackDialog(projectRule.project)
+            val panel = dialog.getViewForTesting()
+
+            assertThat(panel.sentiment).isEqualTo(Sentiment.POSITIVE)
+
+            panel.clearSentimentSelection()
+            assertThat(panel.sentiment).isEqualTo(null)
+            assertThat(dialog.doValidate()).isInstanceOfSatisfying(ValidationInfo::class.java) {
+                it.message.contains(message("feedback.validation.no_sentiment"))
+            }
+        }
+    }
+
+    @Test
+    fun noCommentSet() {
+        runInEdtAndWait {
+            val dialog = FeedbackDialog(projectRule.project)
+            val panel = dialog.getViewForTesting()
+
+            listOf(
```

Comment:
Is this Junit5? Can you make this into a parameterized test?

As it stands, if this test fails it will just say "expected "blah" to contain "empty message" without actually saying which of the listed cases failed.

Reply:
A bit verbose in JUnit 4.12 but made it work.

```
software.aws.toolkits.jetbrains.ui.feedback.FeedbackTest$NonParameterizedTests
  Test noSentimentSet PASSED (8.2s)
  Test commentTooLong PASSED
  Test panelInitiallyPositive PASSED
  Test panelInitiallyNegative PASSED
software.aws.toolkits.jetbrains.ui.feedback.FeedbackTest$NoCommentSetTest
  Test noCommentSet[empty string] PASSED
  Test noCommentSet[spaces] PASSED
  Test noCommentSet[new line] PASSED
  Test noCommentSet[fail] FAILED
```